src/serac/physics/contact/contact_data.cpp

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2019-2023, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/contact/contact_data.hpp"
+
+#include "axom/slic.hpp"
+
+#ifdef SERAC_USE_TRIBOL
+#include "tribol/interface/tribol.hpp"
+#include "tribol/interface/mfem_tribol.hpp"
+
+namespace serac {
+
+ContactData::ContactData(const mfem::ParMesh& mesh)
+    : mesh_{mesh},
+      reference_nodes_{dynamic_cast<const mfem::ParGridFunction*>(mesh.GetNodes())},
+      current_coords_{*reference_nodes_},
+      have_lagrange_multipliers_{false},
+      num_pressure_true_dofs_{0}
+{
+  tribol::initialize(mesh_.SpaceDimension(), mesh_.GetComm());
+}
+
+ContactData::~ContactData() { tribol::finalize(); }
+
+void ContactData::addContactPair(int pair_id, const std::set<int>& bdry_attr_surf1,
+                                 const std::set<int>& bdry_attr_surf2, ContactOptions contact_opts)
+{
+  pairs_.emplace_back(pair_id, mesh_, bdry_attr_surf1, bdry_attr_surf2, current_coords_, contact_opts);
+  if (contact_opts.enforcement == ContactEnforcement::LagrangeMultiplier) {
+    have_lagrange_multipliers_ = true;
+    num_pressure_true_dofs_ += pairs_.back().numPressureTrueDofs();
+  }
+}
+
+bool ContactData::haveContactPairs() const { return !pairs_.empty(); }
+
+void ContactData::update(int cycle, double time, double& dt)
+{
+  tribol::updateMfemParallelDecomposition();
```

Comment:
Is this the active set search?

Reply:
Active set search is here: 
https://github.com/LLNL/serac/blob/516c43ca5b29ce8030e04aaaff1dc749527c24a2/src/serac/physics/contact/contact_interaction.cpp#L117-L133
It's called when determining values for gap/pressure/Jacobian